xonsh/main.py

```diff
@@ -252,10 +253,58 @@ def premain(argv=None):
     return args
 
 
+def _failback_to_other_shells(argv):
+    args = None
+    try:
+        args = premain(argv)
+    except Exception:
+        pass
+    # only failback for interactive shell; if we cannot tell, treat it
+    # as an interactive one for safe.
+    if hasattr(args, 'mode') and args.mode != XonshMode.interactive:
+        sys.exit(1)
```

Comment:
sys.exit [is actually raising exception](https://docs.python.org/3.6/library/sys.html#sys.exit). But I guess raising the original one is better here. will update it later. thanks!